sherlock/internal/api/sherlock/ci_runs_procedures_v3_github_info_list.go

```diff
@@ -0,0 +1,50 @@
+package sherlock
+
+import (
+	"github.com/broadinstitute/sherlock/sherlock/internal/authentication"
+	"github.com/broadinstitute/sherlock/sherlock/internal/errors"
+	"github.com/broadinstitute/sherlock/sherlock/internal/models"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+// ciRunsProceduresV3GithubInfoList godoc
+//
+//	@summary		List GitHub info gleaned from CiRuns
+//	@description	List info about GitHub repos and their workflow files as determined by CiRuns from the past 90 days.
+//	@description	This is a useful proxy for figuring out what repos Sherlock probably has access to: workflows listed
+//	@description	here can probably successfully called by a GitHub Actions deploy hook.
+//	@tags			CiRuns
+//	@produce		json
+//	@success		200						{array}		map[string]map[string][]string
+//	@failure		400,403,404,407,409,500	{object}	errors.ErrorResponse
+//	@router			/api/ci-runs/procedures/v3/github-info [get]
+func ciRunsProceduresV3GithubInfoList(ctx *gin.Context) {
+	db, err := authentication.MustUseDB(ctx)
+	if err != nil {
+		return
+	}
+	type result struct {
+		GithubActionsOwner        string
+		GithubActionsRepo         string
+		GithubActionsWorkflowPath string
+	}
+	var results []result
+	if err = db.Model(&models.CiRun{}).
+		Where("created_at >= current_timestamp - '90 days'::interval").
```

Comment:
What's the reasoning behind the 90 days setting? Is it linked to some technical limitation or "reasonable default" kind of number.

Reply:
Arbitrary balance between "if someone deletes or renames a workflow file, it shouldn't be in this endpoint's response forever" and "workflow files should be suggested even if they haven't run recently".

Making this configurable would be a SQL injection vuln and/or require an annoying amount of logic and it just doesn't really matter so I figured it was fine to hardcode